src/box/memtx_engine.cc

```diff
@@ -68,9 +68,6 @@
 /* sync snapshot every 16MB */
 #define SNAP_SYNC_INTERVAL	(1 << 24)
 
-static void
-checkpoint_cancel(struct checkpoint *ckpt);
-
```

Comment:
You should explain in the commit message what you're doing here.

Reply:
Done. Below is the expanded commit message.


We call `gc_shutdown` after all client shutdown fibers is finished. So
shutdown is quite simple, we just need to finish checkpoint system
fibers.

There may be a checkpoint in progress during shutdown. It can by done in
`checkpoint_daemon` fiber or in client fiber. Anyway we just need to
make sure it is cancellable. In case of memtx we run checkpoint in
distinct cord but thanks to `cord_cojoin` semantics cancel is propagated
to the cord's main fiber. So let's add a yield to the `checkpoint_f` to
make it cancellable.

Now we don't need `checkpoint_cancel` at all. At the moment of
`memtx_engine_shutdown` (actually freeing resources) the cord is sure to
be terminated.